cleverhans/attacks_eager.py

```diff
@@ -0,0 +1,331 @@
+from abc import ABCMeta
+import numpy as np
+from six.moves import xrange
+import warnings
+import collections
+import tensorflow as tf
+from distutils.version import LooseVersion
+import cleverhans.utils as utils
+
+_logger = utils.create_logger("cleverhans.attacks_eager")
+
+if LooseVersion(tf.__version__) < LooseVersion('1.8.0'):
+    error = ('For eager execution',
+             'use Tensorflow version greather than 1.8.0.')
+    print(error)
+    exit(1)
+
+try:
+    from cleverhans.model_eager import Model_Eager
+except ImportError:
+    error = 'Import Error: cleverhans.model_eager'
+    raise ImportError(error)
+
+
+class Attack(object):
```

Comment:
I don't think you should redefine the base class Attack. May be it's an opportunity to refactor the base class from attacks.py so that you can inherit from it to form a new class (say AttackTFE).

Reply:
I agree, 
few options i could think of, let me know your thoughts on which one looks best .... 
Option 1:
base class Attack uses feed_dict and runs through session, inheriting AttackTFE from Attack will bring graph based code structure to AttackTFE, but we can overload the functions, 

PRO: this way no changes are required to existing code base,  
CONS: unnecessary graph def code inherited to AttackTFE, not elegant.

Option 2:
Add a new base class Attack_BASE to contain functions common to Attack and AttackTFE, then inherit both these classes from Attack_BASE.
PRO: hierarchical structure is maintained.
CON: few changes to the existing code base.